src/SourceBuild/tarball/content/prep.sh

```diff
@@ -57,38 +56,74 @@ then
 fi
 
 # Check if Private.SourceBuilt artifacts archive exists
-if [ -f $SCRIPT_ROOT/packages/archive/Private.SourceBuilt.Artifacts.*.tar.gz ]; then
+artifactsBaseFileName="Private.SourceBuilt.Artifacts"
+if [ -f $SCRIPT_ROOT/packages/archive/$artifactsBaseFileName.*.tar.gz ]; then
     echo "  Private.SourceBuilt.Artifacts.*.tar.gz exists...it will not be downloaded"
     downloadArtifacts=false
 fi
 
 # Check if Private.SourceBuilt prebuilts archive exists
-if [ -f $SCRIPT_ROOT/packages/archive/Private.SourceBuilt.Prebuilts.*.tar.gz ]; then
+prebuiltsBaseFileName="Private.SourceBuilt.Prebuilts"
+if [ -f $SCRIPT_ROOT/packages/archive/$prebuiltsBaseFileName.*.tar.gz ]; then
     echo "  Private.SourceBuilt.Prebuilts.*.tar.gz exists...it will not be downloaded"
     downloadPrebuilts=false
 fi
 
+function DownloadArchive {
+    archiveType="$1"
+    isRequired="$2"
+
+    blobStorageUrl="https://dotnetcli.azureedge.net/source-built-artifacts"
+    sourceBuiltArtifactsTarballUrl="${blobStorageUrl}/assets/"
+    sourceBuiltSDKUrl="${blobStorageUrl}/sdks/"
+
+    packageVersionsPath="$SCRIPT_ROOT/eng/Versions.props"
+    notFoundMessage="No source-built $archiveType found to download..."
+
+    echo "  Looking for source-built $archiveType to download..."
+    archiveVersionLine=`grep -m 1 "<PrivateSourceBuilt${archiveType}PackageVersion>" "$packageVersionsPath" || :`
+    versionPattern="<PrivateSourceBuilt${archiveType}PackageVersion>(.*)</PrivateSourceBuilt${archiveType}PackageVersion>"
+
+    if [[ $archiveVersionLine =~ $versionPattern ]]; then
+        if [ "$archiveType" == "SDK" ]; then
+            filename="dotnet-sdk-${BASH_REMATCH[1]}-centos.8-x64.tar.gz"
```

Comment:
It doesn't feel right that the rid is hard-coded here.

Reply:
Given the UX concerns I mentioned, this hard-coded rid and the change to the archive artifacts text file, I lean towards this should be handled by the pipeline itself.